src/main/java/Duke.java

```diff
@@ -1,10 +1,145 @@
+                    save(tasks);
+                    break;
+                } else if (cmd.equals("list")) {
+                    listItems();
+                } else if (cmd.equals("done")) {
+                    int itemNo = sc.nextInt();
+                    markItemAsDone(itemNo);
+                } else if (cmd.equals("delete")) {
+                    int itemNo = sc.nextInt();
+                    deleteItem(itemNo);
+                } else {
+                    String typeOfEvent = cmd;
+                    String eventDescription = sc.nextLine();
+                    addItem(typeOfEvent, eventDescription);
+                }
+            } catch (InvalidCommandException e) {
+                System.out.println(e.getMessage());
+            } catch (EmptyArgumentException e) {
+                System.out.println(e.getMessage());
+            }
+        }
     }
+
+    private static void save(List<Task> tasks) throws IOException {
+        FileWriter writer = new FileWriter(filePath);
+        for(Task task : tasks) {
+            System.out.println("hi");
+            writer.write(convertToFileFormat(task)+'\n');
+        }
+        writer.close();
+    }
+    private static Task convertToTaskFormat(String line) throws EmptyArgumentException {
+        String[] arr = line.split(Pattern.quote(" | "));
+        Task task;
+        if(arr[0].charAt(0) == 'T') task = new Todos(arr[2].trim());
+        else if(arr[0].charAt(0) == 'E') task = new Event(arr[2].trim());
+        else task = new Deadline(arr[2]);
+        if(arr[1].trim().charAt(0) == '1') task.isCompleted = true;
+        else task.isCompleted = false;
+        return task;
+    }
+    private static String convertToFileFormat(Task task) {
+        String fileString = "";
+        if(task instanceof Todos) fileString += "T | ";
+        else if(task instanceof Deadline) fileString += "D | ";
+        else if(task instanceof Event) fileString += "E | ";
+        if(task.isCompleted) fileString += "1 | ";
+        else fileString += "0 | ";
+        fileString += task.description;
+        return fileString;
+    }
+    private static void load() throws IOException, EmptyArgumentException {
+        try {
+            File file = new File(filePath);
+            Scanner sc = new Scanner(file);
+            while(sc.hasNext()) {
+                String line = sc.nextLine();
+                Task task = convertToTaskFormat(line);
+                tasks.add(task);
+                System.out.println("hi");
+            }
+        }
+        catch(FileNotFoundException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+
+
+    private static void addItem(String typeOfEvent, String eventDescription) throws EmptyArgumentException, InvalidCommandException {
+        if(typeOfEvent.equals("todo")) {
+            tasks.add(new Todos(eventDescription));
+        }
+        else {
+            String[] _eventDescription = eventDescription.split("/");
```

Comment:
Please use better variable names. Perhaps a more specific name relating to Todo?